plugins/modules/smart_proxy.py

```diff
@@ -101,13 +110,53 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+
+        lifecycle_environments = module.foreman_params.pop('lifecycle_environments', None)
+        smart_proxy = module.run()
+
+        payload = {
+            'id': smart_proxy['id'],
+        }
+        entity = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
+
+        if handle_lifecycle_environments:
+            desired_environment_ids = set(lifecycle_environment['id'] for lifecycle_environment in lifecycle_environments)
+            current_environment_ids = set(lifecycle_environment['id'] for lifecycle_environment in entity['results']) if entity else set()
```

Comment:
e.g. here, `if entity` does it refer to the smart proxy or to the "set of LCEs attached to it"? (both work, but the later is not really an entity)

Reply:
gothcha, fixed it :-)